semantic-convention-utils/src/main/java/org/hypertrace/semantic/convention/utils/rpc/RpcSemanticConventionUtils.java

```diff
@@ -430,6 +431,24 @@ private static boolean isRpcResponseBodyTruncated(Map<String, AttributeValue> at
     return Optional.empty();
   }
 
+  public static Optional<String> getGrpcRequestMetadataHost(Event event) {
+    if (event.getAttributes() == null || event.getAttributes().getAttributeMap() == null) {
+      return Optional.empty();
+    }
+
+    Map<String, AttributeValue> attributeValueMap = event.getAttributes().getAttributeMap();
+
+    if (!isRpcSystemGrpc(attributeValueMap)) {
+      return Optional.empty();
+    }
+
+    if (attributeValueMap.get(RPC_REQUEST_METADATA_HOST.getValue()) != null) {
```

Comment:
Let's move `RPC_REQUEST_METATA_HOST.getValue()` as static string var at top. 

```
  private static final String RPC_REQUEST_METADATA_HOST_ATTR = RPC_REQUEST_METADATA_HOST.getValue();
```

Reply:
Repace if block with as below to avoid multiple lookups in map.
```
return Optional.ofNullable(attributeValueMap.get(RPC_REQUEST_METADATA_HOST_ATTR))
        .map(v -> v.getValue());
```

Reply:
resolved